cranelift-codegen/meta/src/shared/instructions.rs

```diff
@@ -2630,6 +2634,44 @@ pub fn define(
         .operands_out(vec![a]),
     );
 
+    let x = &operand("x", Any);
+    let a = &operand_doc("a", AnyTo, "Bits of `x` reinterpreted");
+
+    ig.push(
+        Inst::new(
+            "raw_bitcast",
+            r#"
+        Cast the bits in `x` as a different type of the same bit width.
+
+        This instruction does not change the data's representation but allows
+        data in registers to be used as different types, e.g. an i32x4 as a 
+        b8x16. The only constraint on the result `a` is that it can be 
+        `raw_bitcast` back to the original type. Also, in a raw_bitcast between
+        vector types with the same number of lanes, the value of each result 
+        lane is a raw_bitcast of the corresponding operand lane. TODO there is 
+        currently no mechanism for enforcing the bit width constraint.
```

Comment:
Maybe special case in the legalizer? It has more instructions special cased.

Reply:
I think I mean a constraint mechanism on `TypeVar`: like there eventually should be a way to create a two `TypeVar`s, A and B, in this file and constrain B so that it can be a different type than A but it must have the same number of bits. Am I understanding this right, @sunfishcode?

Reply:
@sunfishcode to open a new issue for a mechanism to do this.

Reply:
I've now filed https://github.com/CraneStation/cranelift/issues/854 to track this.